### ISSUE TITLE
there should be a device message in case of SMTP authentication failure

### ISSUE THREAD

**Issue by agutierrez25249** [deltachat/deltachat-android#1218]
- Android version: 9
- Device: xiaomi
- Delta Chat version: 1.0.3
- Expected behavior:
- Actual behavior:
- Steps to reproduce the problem:

I was able to check the email, send and receive without any problem, through the web client (Chrome Browser)

. Reset the application, this time did not solve the problem.
. Reset the connection, solved the problem.

In case it may interest:
The type of connection used, only allows you to check email, that is, there is no internet access.

I hope this information is helpful to continue improving this magnificent application.
- Screenshots:
![Screenshot_2020-01-19-17-42-31-197_chat delta](https://user-images.githubusercontent.com/52469394/72749923-62101180-3b89-11ea-93dc-e51877f2bfd5.jpg)

- Logs:
[deltachat-log-20200119-174930.txt](https://github.com/deltachat/deltachat-android/files/4087599/deltachat-log-20200119-174930.txt)

<!--

**Comment by r10s** [comment]
regarding the log, the error seemed to be ` Respuesta de imap.nauta.cu:143: could not resolve address ("imap.nauta.cu", 143)` - not sure, if this was a delta error. at least delta seems to retry every some minutes.

**Comment by agutierrez25249** [comment]
I often encounter connection problems, I thought I had no connection, but it occurred to me to open my webmail from the same mobile and with the same connection, and it worked very well, Delta still could not send, despite resetting many times, just sent the message, when I reestablished the connection. After that I did the log capture.

**Comment by angelo-fuchs** [comment]
Looking through the log it seems that there was no network for most of the active time
(App start at day 18, 10:30, Network up at day 19, 17:46)
Smtp drops some jobs? (Smtp removes job #6, action SendMsgToSmtp as it cannot be retried)
but at the some time Imap receives messages (so, even though a DNS lookup seemed failing at first, it then started receiving seconds later)
In the log I cannot see a message being sent at all, maybe we should check why the smtp job decided that the sending cannot be retried.

PS: Also the log contains a notification failed exception that I have not seen before starting in Line 877.

**Comment by agutierrez25249** [comment]
I think here you should show the problem much better, 
I hope to be helpful

![Screenshot_2020-01-22-19-27-03-113_chat delta](https://user-images.githubusercontent.com/52469394/73122729-38beff00-3f56-11ea-87a3-fd9f0690e58c.jpg)
Log:
[deltachat-log-20200122-192930.txt](https://github.com/deltachat/deltachat-android/files/4112123/deltachat-log-20200122-192930.txt)

**Comment by gerryfrancis** [comment]
@agutierrez25249 You have been using version 1.1.0, could you update to 1.1.2, perform that test again and report back, please? Thanks.

**Comment by angelo-fuchs** [comment]
@r10s When I read the logfile it seems like the connection was only tried at about 19.27 for the first time, but the screenshot shows messages arriving earlier then that. It looks like the log does not contain the earlier messages, but I don't know why that would be. (Line 42 is start of system, Line 1263 is start of main. Several days later.? Also with enough ui interaction that suggests regular use has been performed.)

**Comment by agutierrez25249** [comment]
the issues still remain in v1.6.0
![deltachat-2020-05-04-132242](https://user-images.githubusercontent.com/52469394/80994462-b2467300-8e0a-11ea-91c0-238e5364557b.jpg)


Log
[deltachat-log-20200503-073528.txt](https://github.com/deltachat/deltachat-android/files/4576159/deltachat-log-20200503-073528.txt)

**Comment by angelo-fuchs** [comment]
Issue in the last one is DNS
From the Log
`failed to lookup address information: No address associated with hostname`

So at the time you could not send the network was working, DC was trying to send, but could not because it could not resolve the host address.

When this happens to you again, please continue to supply the log so that we can investigate further. That also helps to see if the changes that happen in core fix this or not.

**Comment by csb0730** [comment]
I saw a similar behaviour and because it pointed to the device behaviour as issue too, I want to describe it:

When have activated WLAN (1) the device was connected by that if available. Now under bad conditions another public WLAN (2) appeared at the device. The device activated that (2) but had not really access to that. At the end device never found back to the home wlan (1) and DC wasn't able to sent any message. In addition mobile data were active all the time! WLAN has priority.

The only way to solve the problem was: Switch off WLAN for some seconds. Then device switched to mobile net and DC sends data immediately. After that switch on WLAN again, (1) gets activated again an all was fine.

If DC is not sending any data you should check in device settings if device currently has an ip address. Maybe there is no one! This check can be done by user: (Samsung S3: "Settings==>Info about device==>Status==>IP-Adress")
You see to at what network device is connected. Ip address shows that.

Maybe this helps?

**Comment by csb0730** [comment]
When dc detects such a network problem (no ip address, no connection possible, dns problem) it could send a message into device chat? Or a status indicator?
This would show user, that there is a problem - outside - of DC !!!

**Comment by hpk42** [comment]
can you retry the "unsent messages" behaviour with the current test APKs? 

- 🍉https://testrun.org/deltachat-fat-debug-1.9.5.apk - can be used in
  addition/next to gplay
- 🍊https://testrun.org/deltachat-gplay-release-1.9.5.apk - will replace
  gplay and normally keeps data

if in doubt, make a backup beforehand!

**Comment by agutierrez25249** [comment]
the problem continues in v1.9.5
![deltachat-2020-06-13-151614](https://user-images.githubusercontent.com/52469394/84696940-45e78500-af1b-11ea-80ad-24615a558475.jpg)

Log:
[deltachat-log-20200613-151332.txt](https://github.com/deltachat/deltachat-android/files/4782441/deltachat-log-20200613-151332.txt)

**Comment by agutierrez25249** [comment]
I have tried in old versions,> 0.500 and the problem remains. on xiaomi devices when the connection does not have internet access, delta is blocked after 5 minutes of inactivity, it can no longer send but can receive.

**Comment by r10s** [comment]
thanks for the log, this is very helpful.

**Comment by r10s** [comment]
i do not really get it. the log reports several crashes, among them:

- a long-tap on an item in the chat led to a crash because the menu could not be loaded from resources
- in the core, `dc_perform_imap_fetch()` is reported to fail - but this function does not exist anymore.

both issues point to some weird installation, maybe a mix of jar and jni and resources, all coming from different versions.

but maybe also something completely different. as said, this looks weird to me.

do you have a chance to test this on a similar device? or uninstall the app and try over with a new installation?

**Comment by gerryfrancis** [comment]
https://github.com/deltachat/deltachat-android/issues/672 seems to be related, it is a Xiaomi device, too.

**Comment by gerryfrancis** [comment]
> in the core, dc_perform_imap_fetch() is reported to fail - but this function does not exist anymore.

@r10s Since when? According to the log file, the crashes occurred on 06-05, but current core version 1.35 was released on 06-10.

**Comment by r10s** [comment]
dc_perform_imap_fetch()  is gone since core 1.34, about 3 weeks ago, end of may. all nightlies and other testing apks do no have this api anymore. it is not released in the stores, however, this will happen the next days. therefore, i am wondering,

**Comment by gerryfrancis** [comment]
> dc_perform_imap_fetch() is gone since core 1.34, about 3 weeks ago, end of may.

@r10s That was released on 05-27, but the latest official Android version 1.8.1 was released on 05-14 (13 days earlier), so it should have the core version 1.33. Therefore I suppose that the crashes we can see here were recorded with this old core version, and that this log was continued later on with the current version 1.35.

**Comment by r10s** [comment]
okay, i see. the log was created with a much newer version and reports failures before async core.

but then, again, this needs retries.

**Comment by gerryfrancis** [comment]
> the log reports several smtp-jobs beeing processed - are at least some messages being sent?

@r10s It seems so according to the log, but their respective delivery status (change from box to circle) is never updated. Note: Received messages had no subject, I assume they were sent from another mail client than Delta Chat.

**Comment by agutierrez25249** [comment]
In this case, the messages I received were sent from the same device, through a web client
V.1.10.3
![deltachat-2020-06-24-152435](https://user-images.githubusercontent.com/52469394/85619419-445f4080-b630-11ea-9f32-b504274964bf.jpg)
[deltachat-log-20200624-152231.txt](https://github.com/deltachat/deltachat-android/files/4827536/deltachat-log-20200624-152231.txt)

**Comment by gerryfrancis** [comment]
> I remind you, this behavior I have only been able to detect on xiaomi devices, with a mail bag.

@agutierrez25249 That is why I wrote about the ROM of your Xiaomi device here https://github.com/deltachat/deltachat-android/issues/1433#issuecomment-646963052 , and why I believe you should try to update its ROM to the current (or at least stable) version. According to Xiaomi, the one your phone is running now is no stable build. (See my post.) Do your other apps run fine?

**Comment by gerryfrancis** [comment]
@agutierrez25249 I have found this in your recent log:

`06-24 15:20:27.553  8912  8944 W DeltaChat: src/job.rs:485: SMTP connection failure: ConnectionSetupFailure(Io(Custom { kind: Other, error: VerboseError { source: Custom { kind: Other, error: "failed to lookup address information: No address associated with hostname" }, message: "could not resolve address `(\"smtp.nauta.cu\", 25)`" } }))
`

Could you please check if the IP addresses for DNS might be set to static in your network configuration, and if "mail bag" from nauta.cu probably requires a different configuration regarding DNS server entries?

**Comment by agutierrez25249** [comment]
> > Les recuerdo que este comportamiento solo lo he podido detectar en dispositivos xiaomi con una bolsa de correo.
> 
> @ agutierrez25249 Es por eso que escribí sobre la ROM de su dispositivo Xiaomi aquí [# 1433 (comentario)](https://github.com/deltachat/deltachat-android/issues/1433#issuecomment-646963052) , y por qué creo que debería intentar actualizar su ROM a la versión actual (o al menos estable). Según Xiaomi, el que está ejecutando su teléfono ahora no es una construcción estable. (Vea mi publicación). ¿Sus otras aplicaciones funcionan bien?

Precisely my last example was on another xiaomi device, which I updated a few days ago. to respond to your suggestion.

**Comment by gerryfrancis** [comment]
@agutierrez25249 What about your DNS configuration? Does "mail bag" require different DNS entries than those you are using now? Please check with your ISP.

**Comment by agutierrez25249** [comment]
> @agutierrez25249 What about your DNS configuration? Does "mail bag" require different DNS entries than those you are using now? Please check with your ISP.

I do not understand much what you say, but I assure you that I do not have to change anything for the use of the application, when I use a mail bag, or when I use internet packages, the mobile configuration is the same.

**Comment by agutierrez25249** [comment]
I have a silly theory, and that is that xiaomi, with the aim of saving battery, blocks new connections, if you see that there is no internet access

**Comment by gerryfrancis** [comment]
> I have a silly theory, and that is that xiaomi, with the aim of saving battery, blocks new connections, if you see that there is no internet access

@agutierrez25249 I have no clue either. Could you check with a "no-Xiaomi" device? That would be helpful.

**Comment by angelo-fuchs** [comment]
@agutierrez25249 Could you use a DNS Lookup app (like this one: [AndroDNS](https://f-droid.org/en/packages/androdns.android.leetdreams.ch.androdns/)) and when the problem occurs do a Lookup of "smtp.nauta.cu" with it?

**Comment by monperrus** [comment]
FYI, hit by this bug on version 1.10.5

**Comment by gerryfrancis** [comment]
@monperrus Could you follow the advice of @angelo-fuchs and do the DNS lookup when you experience this issue, please? Also please submit a log here. Thanks in advance for reporting back.

**Comment by monperrus** [comment]
No problem with DNS.

Note that the mail is well sent, only the indicator is not updated to "check" :heavy_check_mark:

**Comment by gerryfrancis** [comment]
@monperrus Please submit a log file when this occurs. Thanks!

**Comment by monperrus** [comment]
Looked at the log, seen "authentication failed".

The SMTP password was outdated.

So the proper fix would be a message in the conversation "Email cannot be sent due to authentication failure".

**Comment by r10s** [comment]
> So the proper fix would be a message in the conversation "Email cannot be sent due to authentication failure".

afaik, this is implemented meanwhile, there should be an device-message, wondering why this does not work in this case, cc @Hocuri

**Comment by Hocuri** [comment]
@r10s Only if IMAP login fails. I just assumed that IMAP and SMTP password are the same.

Should be possible to do the same for SMTP, though.

**Comment by gerryfrancis** [comment]
> I just assumed that IMAP and SMTP password are the same.

@Hocuri It is possible to use a different password for the SMTP server in Delta Chat.

**Comment by Hocuri** [comment]
I know, but I didn't think anyone would do this :wink:

**Comment by monperrus** [comment]
OK, AFAIU, the take away is that "there should be a device message in case of SMTP authentication failure". Do we rename the issue accordingly?

**Comment by Hocuri** [comment]
Yes.

**Comment by hpk42** [comment]
@monperrus or @hocuri -- could you maybe rather create a fresh new issue, as a summary of what needs to be done? 
Otherwise lots of people will have to read 41 comments now to make sense of what this issue is about.   Creating a fresh
new one with a nice summary, saying it obsoletes this #1218 one here, will properly cross-link think and somebody wanting to take it on has a much easier start.

**Comment by Hocuri** [comment]
Yes.

**Comment by monperrus** [comment]
Ack, #1508 is now the reference place to discuss this issue.
